src/overlays/actors/ovl_En_Butte/z_en_butte.c

```diff
@@ -21,41 +45,373 @@ const ActorInit En_Butte_InitVars = {
+        ((Math3D_Dist2DSq(player->actor.posRot.pos.x, player->actor.posRot.pos.z, this->actor.initPosRot.pos.x,
+                          this->actor.initPosRot.pos.z) < SQ(120.0f)) ||
+         (this->actor.xzDistFromLink < 60.0f))) {
+        EnButte_SetupFollowLink(this);
+        this->unk_257 = 2;
+    } else if (this->actor.xzDistFromLink < 120.0) {
+        this->unk_257 = 1;
+    } else {
+        this->unk_257 = 0;
+    }
+}
+
+void EnButte_SetupFollowLink(EnButte* this) {
+    EnButte_SelectFlightParams(this, &sFollowLinkParams[this->flightParamsIdx]);
+    this->actionFunc = EnButte_FollowLink;
+}
+
+void EnButte_FollowLink(EnButte* this, GlobalContext* globalCtx) {
+    static s32 D_809CE410 = 1500;
+
+    EnButteFlightParams* flightParams = &sFollowLinkParams[this->flightParamsIdx];
+    Player* player = PLAYER;
+    f32 distFromHome;
+    Vec3f swordTip;
+    f32 animSpeed;
+    f32 minAnimSpeed;
+    f32 distFromSword;
+    s16 yaw;
+
+    func_809CD634(this);
+    Math_SmoothScaleMaxMinF(&this->actor.speedXZ, flightParams->speedXZTarget, flightParams->speedXZScale,
+                            flightParams->speedXZStep, 0.0f);
+    minAnimSpeed = 0.0f;
+
+    if ((this->flightParamsIdx != 0) && (this->timer < 12)) {
+        swordTip.x = player->swordInfo[0].tip.x + Math_Sins(player->actor.shape.rot.y) * 10.0f;
+        swordTip.y = player->swordInfo[0].tip.y;
+        swordTip.z = player->swordInfo[0].tip.z + Math_Coss(player->actor.shape.rot.y) * 10.0f;
+
+        yaw = Math_Vec3f_Yaw(&this->actor.posRot.pos, &swordTip) + (s16)(Math_Rand_ZeroOne() * D_809CE410);
+        if (Math_ApproxUpdateScaledS(&this->actor.posRot.rot.y, yaw, 2000) != 0) {
+            if (globalCtx->gameplayFrames % 2) {
+                this->actor.posRot.rot.y += (s16)(sinf(this->unk_25C) * 60.0f);
+            }
+        } else {
+            minAnimSpeed = 0.3f;
+        }
+    }
+
+    this->posYTarget = MAX(player->actor.posRot.pos.y + 30.0f, player->swordInfo[0].tip.y);
+
+    EnButte_Turn(this);
+
+    animSpeed = this->actor.speedXZ / 2.0f + Math_Rand_ZeroOne() * 0.2f + (1.0f - Math_Sins(this->unk_260)) * 0.15f +
+                (1.0f - Math_Sins(this->unk_25E)) * 0.3f + minAnimSpeed;
+    this->skelAnime.animPlaybackSpeed = CLAMP(animSpeed, 0.2f, 1.5f);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Butte/func_809CD110.s")
+    if (this->timer <= 0) {
+        EnButte_SelectFlightParams(this, &sFollowLinkParams[this->flightParamsIdx]);
+        D_809CE410 = -D_809CE410;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Butte/func_809CD144.s")
+    distFromHome = Math3D_Dist2DSq(this->actor.posRot.pos.x, this->actor.posRot.pos.z, this->actor.initPosRot.pos.x,
+                                   this->actor.initPosRot.pos.z);
+    if ((player->heldItemActionParam != 6) || !(fabsf(player->actor.speedXZ) < 1.8f) || (this->swordDownTimer > 0) ||
```

Comment:
With every term of this conditional being negated, it might look better as `if (!((player->heldItemActionParam == 6) && (fabsf(player->actor.speedXZ) < 1.8f) && (this->swordDownTimer <= 0) && (distFromHome < SQ(320.0f))))`.